File: Algorithms/src/1658. Minimum Operations to Reduce X to Zero/Code/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int x) {
        long long total = 0;
        for (int num : nums) total += num;
        if (total < x) return -1;
        long long target = total - x;
        if (target == 0) return nums.size();
        int n = nums.size();
        int left = 0;
        long long sum = 0;
        int maxLen = -1;
        for (int right = 0; right < n; ++right) {
            sum += nums[right];
            while (sum > target && left <= right) {
                sum -= nums[left];
                ++left;
            }
            if (sum == target) {
                maxLen = max(maxLen, right - left + 1);
            }
        }
        return maxLen == -1 ? -1 : n - maxLen;
    }
};

File: Algorithms/src/1658. Minimum Operations to Reduce X to Zero/Code/solution.py
class Solution:
    def minOperations(self, nums: list[int], x: int) -> int:
        total = sum(nums)
        if total < x:
            return -1
        target = total - x
        if target == 0:
            return len(nums)
        n = len(nums)
        left = 0
        curr = 0
        max_len = -1
        for right in range(n):
            curr += nums[right]
            while curr > target and left <= right:
                curr -= nums[left]
                left += 1
            if curr == target:
                max_len = max(max_len, right - left + 1)
        return -1 if max_len == -1 else n - max_len